runtime/lua/vim/_editor.lua

```diff
@@ -892,6 +892,108 @@ function vim._init_default_mappings()
   ]])
 end
 
+--- Attempt to read the file at {path} prompting the user if the file should be
+--- trusted. The user's choice is persisted in a trust database at
+--- $XDG_STATE_HOME/nvim/trust.
+---
+---@param path (string) Path to a file to read.
+---@param opts (table|nil) Options (currently unused).
+---
+---@return (string|nil) The contents of the given file if it is trusted, or nil
+---                     otherwise.
+function vim.readsecure(path, opts)
+  local fullpath = vim.fn.fnamemodify(vim.fs.normalize(path), ':p')
+  local trust = {}
+  do
+    local f = io.open(vim.fn.stdpath('state') .. '/trust', 'r')
+    if f then
+      local contents = f:read('*a')
+      if contents then
+        for line in vim.gsplit(contents, '\n') do
+          local hash, file = string.match(line, '^(%S+) (.+)$')
+          if hash and file then
+            trust[file] = hash
+          end
+        end
+      end
+      f:close()
+    end
+  end
+
+  if trust[fullpath] == '!' then
+    -- File is blacklisted
+    return nil
+  end
+
+  local contents
+  do
+    local f = io.open(fullpath, 'r')
+    if not f then
+      return nil
+    end
+    contents = f:read('*a')
+    f:close()
+  end
+
+  local hash = vim.fn.sha256(contents)
+  if trust[fullpath] == hash then
+    -- File already exists in trust database
+    return contents
+  end
+
+  -- File either does not exist in trust database or the hash does not match
+  local choice = vim.fn.confirm(
+    string.format('%s is not trusted.', fullpath),
+    '&ignore\n&view\n&blacklist\n&trust',
+    1
+  )
+
+  if choice == 0 or choice == 1 then
+    -- Cancelled or ignored
+    return nil
+  elseif choice == 2 then
+    -- View
+    local buf = vim.api.nvim_create_buf(false, true)
+    local lines = vim.split(contents, '\n')
+
+    -- Trim trailing newline
+    if lines[#lines] == '' then
+      lines[#lines] = nil
+    end
+
+    vim.api.nvim_buf_set_lines(buf, 0, -1, false, lines)
+    vim.cmd('new')
+    local win = vim.api.nvim_get_current_win()
+    vim.api.nvim_win_set_buf(win, buf)
```

Comment:
```suggestion
    vim.api.nvim_set_current_buf(buf)
```

Reply:
Fixed in a1dc757291aed157943cc0ac94a634acc2f0d40c